modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/treeview/NodeList.js

```diff
@@ -44,7 +44,12 @@ export default function NodeList({
 			}}
 			onFocus={(event) => {
 				if (onFocus) {
-					onFocus(event);
+					onFocus(event, {useKeyboardNavigation: true});
+				}
+			}}
+			onMouseEnter={(event) => {
+				if (onFocus) {
+					onFocus(event, {useKeyboardNavigation: false});
```

Comment:
This is not a great separation of concerns. A `mouseEnter` event should invoke an `onMouseEvent` callback. A higher component may then do the same action on both events.

Reply:
Hi @markocikos,

I understand your point of view, it is true that if you click on the item the onFocus would be called as well but already will be initialized by mouseEnter.

What I did was more simple, using onMouseDown to prevent onFocus to even being called when mouse is used to select a node list item.

Let me know what do you think about this approach.

Regards.